workflows: instructions for release process

amtep opened the pull request:
## What, How & Why?
Describes the new release process that uses Github Actions.

Please let me know if any step is still unclear.

## ☑️ ToDos
<!-- Add your own todos here -->
* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 🚦 Tests
* [ ] 🔀 Executed flexible sync tests locally if modifying flexible sync
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

kneth reviewed (approved):
LGTM. I am looking forward to release using GHA.

kraenhansen commented on contrib/how-to-release.md:
```diff
@@ -1,4 +1,38 @@
-# How to release Realm JavaScript
+# How to release Realm JavaScript (with Actions)
+
+We're moving toward releasing via GitHub Actions, but since the Actions can't yet do prereleases, this document contains both the new way and the old way.
```
They can now.

amtep commented:
Since we now have prereleases with github actions, I removed the old instructions completely.

I also updated the instructions for the new input fields in prepare release.
